index.js

```diff
@@ -112,6 +107,13 @@
     };
   }
 
+  //  clean :: Object -> Object
+  function clean(x) {
+    var k, o = Object.create(null);
+    for (k in x) o[k] = x[k];
```

Comment:
An own-property check here would be reasonable.

Also, I like to pretend we have block-scoped variables, so I would prefer `for (var k in x)` since `k` is only referenced in the loop.